spark/spark-shims/src/main/java/org/apache/zeppelin/spark/SparkShims.java

```diff
@@ -58,7 +58,7 @@ public SparkShims(Properties properties) {
   private static SparkShims loadShims(int sparkMajorVersion, Properties properties, Object entryPoint)
       throws Exception {
     Class<?> sparkShimsClass;
-    if (sparkMajorVersion == 3) {
+    if (sparkMajorVersion == 3 || sparkMajorVersion == 4) {
       LOGGER.info("Initializing shims for Spark 3.x");
       sparkShimsClass = Class.forName("org.apache.zeppelin.spark.Spark3Shims");
```

Comment:
The current shims appear to still work fine. I'm not sure if we want to rename the shim, or remove the shim since different functionality isn't needed right now

Reply:
Yes, I think we should remove the shim layer to simplify the codebase given the Spark API is pretty stable in recent versions, the corner compatibility cases could be resolved by using reflection directly.

Reply:
Sounds good, I'll work on that

Reply:
I've been trying to remove the shims and consolidate things but I keep running into class loading errors and I'm not really sure what to do about it. I moved most from the shim into a `SparkInterpreterUtils` class in a `spark-common` module, but just keep running into

```
IllegalAccess failed to access class org.apache.zeppelin.spark.SparkInterpreterUtils from class org.apache.zeppelin.spark.SparkScala213Interpreter (org.apache.zeppelin.spark.SparkInterpreterUtils is in unnamed module of loader 'app'; org.apache.zeppelin.spark.SparkScala213Interpreter is in unnamed module of loader java.net.URLClassLoader @2c768ada)
```

Reply:
the error indicates they are loaded by different classloaders ... classloader issues are always quirky, and I may not be able to give more advice

Reply:
Started from scratch and got things mostly deduped, renaming shims to common

Reply:
Spark community proposes postponing the 4.0.0 release ETA to 2025, could you please make a dedicated PR for shims remove? we can merge that earlier.

Reply:
It would still be nice to be able to use the preview releases in Zeppelin which this enables

Reply:
I mean the only things actually related to Spark 4 in this are:
- Maven profile which will be helpful to be able to run manual tests against the preview release
- CI stage for testing that will have to be removed or commented out since it's not an officially supported version

So if/when you're good with everything else, I can just remove the CI stage (preferably keep the profile in), and it will mostly be a shim removal PR that just happens to remove the requirement that the major version is 3

Reply:
sgtm

Reply:
Changed those back and update the title and description